fe/fe-core/src/main/java/org/apache/doris/nereids/types/DateTimeV2Type.java

```diff
@@ -78,6 +78,17 @@ public static DateTimeV2Type forType(DataType dataType) {
         return MAX;
     }
 
+    /**
+     * return proper type of datetimev2 for VarcharLiteral
+     * may be we need to check for validity?
+     */
+    public static DateTimeV2Type forTypeFromVarcharLiteral(String s) {
```

Comment:
```suggestion
    public static DateTimeV2Type forExpression(Expression e) {
        if (e instanceof StringLikeLiteral) {
            ....
        } else {
            return forType(e.getDataType());
        }
```

Reply:
It has already been determined StringLikeLiteral, so it is unnecessary to check again.